lib/util/utils.js

```diff
@@ -21,5 +25,46 @@ exports.matchMemberExpression = function (name, node) {
       return true;
     }
   }
+
   return match(node.object, nameParts.length - 2);
 };
+
+/**
+ * Matches a MemberExpression by traversing the node and returns true if any of the property names match.
+ * @param nameRegex
+ * @param node
+ * @returns {Boolean}
+ */
+exports.matchRegexMemberExpression = function matchRegexMemberExpression(nameRegex, node) {
+  if (!node) {
+    return false;
+  }
+
+  if (node.type === 'MemberExpression') {
+    if (nameRegex.test(node.property.name)) {
+      return true;
+    }
+
+    return matchRegexMemberExpression(nameRegex, node.object);
+  }
+
+  return !!(node.type === 'Identifier' && nameRegex.test(nameRegex));
+};
+
+/**
+ * Matches a string to a possible regex.
+ * @param {String|RegExp} toMatch
+ * @param {String} other
+ * @returns {Boolean}
+ */
+exports.match = function (toMatch, other) {
+  if (exports.isRegEx(toMatch)) {
+    return toMatch.test(other);
+  }
+
+  return toMatch === other;
+};
+
+exports.isRegEx = function (possibleRegEx) {
```

Comment:
Wasn't sure about adding another dependency but have gone ahead and switched to using that instead!